lib/src/data/category.dart

```diff
@@ -0,0 +1,37 @@
+import 'types.dart';
+
+class Category {
+  // Stores the unique ID for this category
+  final CategoryID categoryID;
+  // Stores the name of the category
+  final String title;
+  // Stores a brief description for the category
+  final String description;
+  // Stores the link of the thumbnail for this category
+  final String imagePath;
+  // When none of the products of the category are available, the category is still shown since the Product is not delisted. If the seller wants to get rid of the listing, they can simply set this to 'true' to hide this particular category from buyers.
+  final bool delisted;
```

Comment:
Yes! I think the delisted for category is redundant!
Here are my two thoughts for this change:

1) We can show no products for empty categories . (I'm less inclined to this because of the user experience)
2) While fetching the categories, I can add another check that the product from the seller should have a false value for delisted. Only then add its category. (I'm more inclined to this)

Let me know your thoughts on this.